src/utils.ts

```diff
@@ -0,0 +1,221 @@
+ */
+export 
+function getBodyData(key: string): string {
+  var val = document.body.getAttribute(key);
+  if (val) {
+    return decodeURIComponent(val);
+  }
+  return null;
+}
+
+
+/**
+ * Join a sequence of url components with '/',
+ * encoding each component with encodeURIComponent.
+ */
+export
+function urlJoinEncode(...args: string[]): string {
+  return encodeURIComponents(urlPathJoin.apply(null, args));
+}
+
+
+/**
+ * Properly detect the current browser.
+ * http://stackoverflow.com/questions/2400935/browser-detection-in-javascript
+ */
+export
+var browser: string[] = (() => {
+  if (typeof navigator === 'undefined') {
+    // navigator undefined in node
+    return ['None'];
+  }
+  var N: string = navigator.appName;
+  var ua: string = navigator.userAgent
+  var tem: RegExpMatchArray;
+  var M: RegExpMatchArray = ua.match(/(opera|chrome|safari|firefox|msie)\/?\s*(\.?\d+(\.\d+)*)/i);
+  if (M && (tem = ua.match(/version\/([\.\d]+)/i)) !== null) M[2] = tem[1];
+  M = M ? [M[1], M[2]] : [N, navigator.appVersion, '-?'];
+  return M;
+})();
+
+
+/** 
+ * Return a serialized object string suitable for a query.
+ *
+ * http://stackoverflow.com/a/30707423
+ */
+export
+function jsonToQueryString(json: any): string {
+  return '?' +
+    Object.keys(json).map((key: string): any => {
+      return encodeURIComponent(key) + '=' +
+        encodeURIComponent(json[key]);
+    }).join('&');
+}
+
+
+/**
+ * Input settings for an AJAX request.
+ */
+export
+interface IAjaxSetttings {
+  method: string;
+  dataType: string;
+  contentType?: string;
+  data?: any;
+}
+
+
+/*
+ * Asynchronous XMLHTTPRequest handler.
+ *
+ * http://www.html5rocks.com/en/tutorials/es6/promises/#toc-promisifying-xmlhttprequest
+ */
+export
+function ajaxRequest(url: string, settings: IAjaxSetttings): Promise<any> {
+  return new Promise((resolve, reject) => {
+    var req = new XMLHttpRequest();
+    req.open(settings.method, url);
+    if (settings.contentType) {
+      req.overrideMimeType(settings.contentType);
+    }
+
+    req.onload = () => {
+      if (req.status == 200) {
+        if (settings.dataType === 'json') {
+          resolve(JSON.parse(req.response));
+        } else {
+          resolve(req.response);
+        }
+      } else {
+        reject(req.statusText);
+      }
+    }
+
+    req.onerror = () => {
+      reject(req.statusText);
+    }
+
+    if (settings.data) {
+      req.send(settings.data);
+    } else {
+      req.send();
+    }
+  });
+}
+
+
+/**
+ * Stub for require.js module loader.
+ */
+declare
+function require(modules: string[], success: Function, reject?: Function): void;
+
+
+/**
+ * Try to load a class.
+ *
+ * Try to load a class from a module using require.js, if a module 
+ * is specified, otherwise tries to load a class from the global 
+ * registry, if the global registry is provided.
+ */
+export
+function loadClass(class_name: string, module_name: string, registry: { [string: string]: Function; }) : Promise<string> {
```

Comment:
is this a require things ? I don't see it use here (I guess it will be usefull later)